### PR TITLE
[DI] add id of referencing service when a deprecated alias is found

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Improves #29968 a bit for DX.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Nice improvement! Not sure if it's possible, but I'd reword the error message:

```
As referenced by "alias": the "deprecated_foo_alias" service alias
is deprecated. You should stop using it, as it will soon be removed.
```

like this:

```
The "deprecated_foo_alias" service alias referenced by "alias"
is deprecated. You should stop using it, as it will soon be removed.
```

**Comment by nicolas-grekas** [comment]
I'd happily improve the message, but your suggestion is not possible, see L65 how the message is constructed. We can only append or prepend to the message conveyed by the alias.

**Review by javiereguiluz — APPROVED** [review]
Nice!

**Comment by stof** [comment]
Note that this PR is renaming the placeholder used in the message. When it gets merged, the blog post published today should be updated to reflect the change.

**Review by stof — APPROVED** [review]

**Comment by stof** [comment]
@nicolas-grekas with your change to move the trigger_error outside the loop, what happens if you have `non-deprecated alias -> deprecated alias -> actual service` ? Do we still get a warning somewhere (so that the author of `non-deprecated alias` can know that they should update their alias) ?

**Comment by nicolas-grekas** [comment]
> Do we still get a warning somewhere (so that the author of non-deprecated alias can know that they should update their alias) ?

yes, because of the loop in process (see L32)

**Comment by javiereguiluz** [comment]
@stof thanks for this warning. The blog post has been updated accordingly.
